app/controllers/locations_controller.rb

```diff
@@ -46,6 +52,19 @@ def update
 
 private
 
+  def valid_date_params?(location_params)
+    is_integer?(location_params["startdate(1i)"]) && is_integer?(location_params["startdate(2i)"]) && is_integer?(location_params["startdate(3i)"]) &&
+      Date.valid_date?(location_params["startdate(1i)"].to_i, location_params["startdate(2i)"].to_i, location_params["startdate(3i)"].to_i)
+  end
+
+  def date_params_missing?(location_params)
+    location_params["startdate(1i)"].blank? && location_params["startdate(2i)"].blank? && location_params["startdate(3i)"].blank?
```

Comment:
Isn't it enough for any one of these to be blank? (i.e. || instead of &&)

Reply:
Good spot! One is enough yup